pkgs/development/interpreters/python/cpython/2.7/default.nix

```diff
@@ -55,6 +55,13 @@ let
 
       ./properly-detect-curses.patch
 
+      # Patch from http://bugs.python.org/issue1222585 adapted to work with
+      # `patch -p1' and with last hunk removed
+      # Upstream distutils is calling C compiler to compile C++ code, which
+      # only works for GCC and Apple Clang. This makes distutils to call C++
+      # compiler when needed.
+      ./python-2.7-distutils-C++.patch
+
```

Comment:
Is this patch accepted upstream? How does this affect non-Darwin? Does this fix anything on non-Darwin?